server/src/test/java/org/apache/cassandra/sidecar/handlers/NodeDrainHandlerTest.java

```diff
@@ -0,0 +1,243 @@
+              .expect(ResponsePredicate.SC_OK)
+              .send(context.succeeding(response -> {
+                  assertThat(response.statusCode()).isEqualTo(OK.code());
+                  LOGGER.info("Drain Response: {}", response.bodyAsString());
+
+                  OperationalJobResponse drainResponse = response.bodyAsJson(OperationalJobResponse.class);
+                  assertThat(drainResponse).isNotNull();
+                  assertThat(drainResponse.jobId()).isNotNull();
+                  assertThat(drainResponse.operation()).isEqualTo(EXPECTED_OPERATION_NAME);
+                  context.completeNow();
+              }));
+    }
+
+    @Test
+    void testDrainConflictWhenDraining(VertxTestContext context)
+    {
+        when(mockStorageOperations.operationMode()).thenReturn(OPERATION_MODE_DRAINING);
+
+        WebClient client = WebClient.create(vertx);
+        client.put(server.actualPort(), "127.0.0.1", TEST_ROUTE)
+              .expect(ResponsePredicate.SC_CONFLICT)
+              .send(context.succeeding(response -> {
+                  assertThat(response.statusCode()).isEqualTo(CONFLICT.code());
+                  LOGGER.info("Drain Response: {}", response.bodyAsString());
+
+                  OperationalJobResponse drainResponse = response.bodyAsJson(OperationalJobResponse.class);
+                  assertThat(drainResponse).isNotNull();
+                  assertThat(drainResponse.jobId()).isNotNull();
+                  assertThat(drainResponse.operation()).isEqualTo(EXPECTED_OPERATION_NAME);
+                  context.completeNow();
+              }));
+    }
+
+    @Test
+    void testDrainAllowedWhenJoining(VertxTestContext context)
+    {
+        when(mockStorageOperations.operationMode()).thenReturn(OPERATION_MODE_JOINING);
+
+        WebClient client = WebClient.create(vertx);
+        client.put(server.actualPort(), "127.0.0.1", TEST_ROUTE)
+              .send(context.succeeding(response -> {
+                  assertThat(response.statusCode()).isEqualTo(OK.code());
+                  LOGGER.info("Drain Response: {}", response.bodyAsString());
+
+                  OperationalJobResponse drainResponse = response.bodyAsJson(OperationalJobResponse.class);
+                  assertThat(drainResponse).isNotNull();
+                  assertThat(drainResponse.status()).isEqualTo(SUCCEEDED);
+                  assertThat(drainResponse.jobId()).isNotNull();
+                  assertThat(drainResponse.operation()).isEqualTo(EXPECTED_OPERATION_NAME);
+                  context.completeNow();
+              }));
+    }
+
+    /**
+     * Test guice module for Node Drain handler tests
+     */
+    class NodeDrainTestModule extends AbstractModule
+    {
+        @Provides
+        @Singleton
+        public InstancesMetadata instanceMetadata()
+        {
+            final int instanceId = 100;
+            final String host = "127.0.0.1";
+            final InstanceMetadata instanceMetadata = mock(InstanceMetadata.class);
```

Comment:
NIT: remove final keyword from local variables
```suggestion
            int instanceId = 100;
            String host = "127.0.0.1";
            InstanceMetadata instanceMetadata = mock(InstanceMetadata.class);
```

Reply:
updated.